compiler/rustc_hir_analysis/src/collect/lifetimes.rs

```diff
@@ -1195,8 +1195,10 @@ impl<'a, 'tcx> LifetimeContext<'a, 'tcx> {
                     // Fresh lifetimes in APIT used to be allowed in async fns and forbidden in
                     // regular fns.
                     if let Some(hir::PredicateOrigin::ImplTrait) = where_bound_origin
-                        && let hir::LifetimeName::Param(_) = lifetime_ref.res
-                        && lifetime_ref.is_anonymous()
+                        && let hir::LifetimeName::Param(param_id) = lifetime_ref.res
+                        && let Some(generics) = self.tcx.hir().get_generics(self.tcx.local_parent(param_id))
+                        && let Some(param) = generics.params.iter().find(|p| p.def_id == param_id)
+                        && param.is_elided_lifetime()
```

Comment:
With this PR, the error message still reads as ""anonymous lifetimes in `impl Trait` are unstable"

should it change to "elided lifetimes" instead of "anonymous lifetimes"?

Reply:
(this feedback need not block the landing of this PR. It was just something I noticed while looking it over)